db_stress_tool/db_stress_common.cc

```diff
@@ -84,9 +151,17 @@ std::vector<int64_t> GenerateNKeys(ThreadState* thread, int num_keys,
   int64_t next_key = base_key + thread->rand.Next() % FLAGS_active_width;
   keys.push_back(next_key);
   for (int i = 1; i < num_keys; ++i) {
-    // This may result in some duplicate keys
-    next_key = next_key + thread->rand.Next() %
-                              (FLAGS_active_width - (next_key - base_key));
+    // Generate the key follows zipfian distribution
+    if (FLAGS_hot_key_alpha != 0) {
+      double float_rand =
+          (static_cast<double>(thread->rand.Next() % FLAGS_max_key)) /
+          FLAGS_max_key;
+      next_key = GetOneHotKeyID(float_rand, FLAGS_max_key);
+    } else {
+      // This may result in some duplicate keys
+      next_key = next_key + thread->rand.Next() %
+                                (FLAGS_active_width - (next_key - base_key));
```

Comment:
It used to be that, the keys are sorted. But randomized keys aren't ordered. Is it a problem?

Reply:
@siying The comment is the original one. From my understanding, the next_key might be duplicate in the array since when the range shrinks (FLAGS_active_width - (next_key - base_key)), the key might be the same when next_key add a rand in the range.

Reply:
Still, it used to be that every key generated is larger than the previous one.
I just wonder whether the user of the function makes assumption about the key ordering.
When you run "make crash_test", does it pass?

Reply:
Or maybe simply run db_stress with this new option.

Reply:
@siying I have add the --hot_key_alpha=0.95 to db_crashtest.py and it passes the verification tests. I checked the code, db_crashtest.py uses an unorderset to hold the keys with no overlap. GenerateNKeys are only used in two cases: 1) Generate an array of keys for TestMultiGet, 2) Generate an array of keys as the N number of start-key for TestIterator. It is ok for them to have an array of keys that are 1) have duplication 2) the keys does not have order. To make it clear, I will add comments to GFLAGS and ahead of these function say:  the keys being generated do not ensure order (current one is always larger(smaller) than previous one).